data_extraction/word_embeddings_reader.py

```diff
@@ -44,7 +45,7 @@ def get_token_embedding_from_full(token):
 
     return full_embeddings_matrix_dict.get(token)
 
-
+@functools.lru_cache(maxsize=5)
```

Comment:
Nice! Did you see a significant speedup?

Reply:
yup. If I call the function multiple times during the same run it's actually only called once. Useful when running more experiments at once